src/librustc/ty/maps.rs

```diff
@@ -107,13 +118,25 @@ impl<'tcx> Value<'tcx> for Ty<'tcx> {
     }
 }
 
-
 impl<'tcx> Value<'tcx> for ty::DtorckConstraint<'tcx> {
     fn from_cycle_error<'a>(_: TyCtxt<'a, 'tcx, 'tcx>) -> Self {
         Self::empty()
     }
 }
 
+impl<'tcx> Value<'tcx> for SymbolName {
+    fn from_cycle_error<'a>(_: TyCtxt<'a, 'tcx, 'tcx>) -> Self {
+        SymbolName { name: Symbol::intern("<error>").as_str() }
+    }
+}
+
+#[derive(Clone)]
+pub struct SymbolName {
+    // FIXME: we don't rely on interning or equality here - better have
+    // this be a `&'tcx str`.
```

Comment:
Actually, that comment was needed because a previous try used `Symbol` there, which as you said is dangerous. I'll remove it.